lib/matplotlib/backend_bases.py

```diff
@@ -2226,6 +2232,9 @@ def print_figure(
                 _bbox_inches_restore = None
 
             try:
+                # we have already done CL above, so turn it off:
+                cl_state = self.figure.get_constrained_layout()
+                self.figure.set_constrained_layout(False)
```

Comment:
I suspect this should not be part of `try`. We're not expecting it to fail.

Also, does it make sense to
~~~
cl_state = self.figure.get_constrained_layout()
if cl_state:
    self.figure.set_constrained_layout(False)

[...]

if cl_state:
    self.figure.set_constrained_layout(cl_state)
~~~

to prevent any overhead if constrained is not used?

Note that this can easily be refactored to a context manager now:
~~~
with self.figure._constrained_layout_deactivated():
    try:
        [...]
    finally:
        [...]
return result
~~~

Reply:
Sure, the set can be outside the `try`.